aesd-char-driver/main.c

```diff
@@ -263,16 +267,140 @@ ssize_t aesd_write(struct file *filp, const char __user *buf, size_t count,
 }
 
 
+loff_t  aesd_llseek(struct file *filp, loff_t offset, int whence)
+{
+    loff_t retval;
+    struct aesd_dev *driver_data;
+    
+    //Checking validity
+    if(!filp->private_data)
+    {
+        retval = -EINVAL;
+        return retval;
+    }
+
+    driver_data = filp->private_data;
+
+
+    //Aquiring mutex
+    retval = mutex_lock_interruptible(&driver_data->aesd_char_mut);
+    if(retval != 0)
+    {
+        PDEBUG("Could not aquire mutex");
+        return retval;
+    }
+
+    retval = fixed_size_llseek(filp,offset,whence, driver_data->circular_buffer.cir_buf_size);
```

Comment:
@kevintom98 think about how you could design this such that you wouldn't need to hold the mutex during the `fixed_size_llseek` call.